lib/ui/elements/layerStyle.mjs

```diff
@@ -0,0 +1,532 @@
+/**
+### /ui/elements/layerStyle
+
+The module exports an object of methods to for the creation of layer style elements.
+
+@requires /ui/layers
+
+@module /ui/elements/layerStyle
+*/
+
+mapp.utils.merge(mapp.dictionaries, {
+  en: {
+    icon_scaling_field: 'Toggle Field Icon Scaling',
+    icon_scaling_label: 'Icon Scale',
+    icon_scaling_cluster: 'Cluster Icon Scale',
+    icon_scaling_cluster_log: 'Cluster Log Scale',
+    icon_scaling_zoom_in: 'Zoom-in Scale',
+    icon_scaling_zoom_out: 'Zoom-out Scale'
+  }
+});
+
+const methods = {
+  panel,
+  hover,
+  hovers,
+  label,
+  labels,
+  opacitySlider,
+  theme,
+  themes,
+  icon_scaling
+}
+
+export default methods
+
+/**
+@function panel
+
+@description
+The panel() style element method creates an array of style.elements from all available style property keys if not implicit.
+
+The elements array will be mapped to return a flat array of HTMLElements as content for the layer.style.view node.
+
+The layer.style.view is returned for the style drawer shown in the default mapp view layer.view.
+
+@param {layer} layer A decorated mapp layer with a style object.
+
+@property {layer-style} layer.style The layer style configuration.
+@property {Array} [style.elements] Array of method keys to order elements in the layer.style.view.
+
+@returns {HTMLElement} The layer.style.view element.
+*/
+function panel(layer) {
+
+  if (!layer.style) return;
+
+  layer.style.elements ??= Object.keys(layer.style)
```

Comment:
Found an issue here, because it looks for object keys if you provide a property like opacitySlider which isn't an object but a boolean it doesn't assign the key to the layer.style.elements object which wont trigger th function on the panel.